### PR TITLE
Add nanoflow event

### PR DESCRIPTION
**Pull request by jseremba**

### CODE REVIEW

File: src/components/SwitchContainer.ts [hunk 1]
```diff
@@ -102,10 +104,11 @@ class SwitchContainer extends Component<SwitchContainerProps, SwitchContainerSta
     }
 
     private handleToggle() {
-        const { booleanAttribute, mxObject, onChangeMicroflow } = this.props;
+        const { booleanAttribute, mxObject, onChangeMicroflow, onChangeNanoflow } = this.props;
         if (mxObject) {
             mxObject.set(booleanAttribute, !mxObject.get(booleanAttribute));
             this.executeAction(onChangeMicroflow, mxObject.getGuid());
+            this.executeAction(onChangeNanoflow, mxObject.getGuid());
```
Comment (djkwagala) [inline src/components/SwitchContainer.ts]: I suppose the `nanoflow (client) should run before the microflow (server)`
Please include conditional not to run a `microflow` while **offline** however the `nanoflow` should still be executed while offline

File: src/components/SwitchContainer.ts [hunk 2]
```diff
@@ -153,7 +156,9 @@ class SwitchContainer extends Component<SwitchContainerProps, SwitchContainerSta
     }
 
     private executeAction(actionname: string, guid: string) {
-        if (actionname && this.props.mxObject) {
+        const { mxObject, mxform } = this.props;
+
+        if (actionname && this.props.mxObject && actionname !== "onChangeNanoflow") {
```
Comment (djkwagala) [inline src/components/SwitchContainer.ts]: `mxObject` is already retrieved from `this.props` via destructuring, so Correct to `if (actionname && mxObject && actionname !== "onChangeNanoflow") `

File: src/components/SwitchContainer.ts [hunk 3]
```diff
@@ -162,6 +167,18 @@ class SwitchContainer extends Component<SwitchContainerProps, SwitchContainerSta
                     guids: [ guid ]
                 }
             });
+        } else if (mxObject) {
+            const context = new mendix.lib.MxContext();
+            context.setContext(mxObject.getEntity(), mxObject.getGuid());
+            window.mx.data.callNanoflow({
+                context,
+                error: (error: Error) => {
+                    mx.ui.error("Error executing nanoflow " + actionname + "onChangeNanoflow" + " : " +
```
Comment (djkwagala) [inline src/components/SwitchContainer.ts]: Hello, use the string literals like in the microflow code Line164
> window.mx.ui.error(\`Error while executing nanoflow ${actionname}: ${error.message}\`)

**Review by djkwagala — CHANGES_REQUESTED** [review]
Please reminder to bump up version and change it in package-lock.json as well.

File: src/components/SwitchContainer.ts (hunk 3 above)
Comment (jseremba) [reply src/components/SwitchContainer.ts]: Done

File: src/components/SwitchContainer.ts [hunk 4]
```diff
@@ -18,6 +19,7 @@ interface SwitchContainerProps extends WrapperProps {
     label: string;
     labelWidth: number;
     onChangeMicroflow: string;
+    onChangeNanoflow: string;
```
Comment (McDoyen) [inline src/components/SwitchContainer.ts]: Is ```onChangeNanoflow``` of type ``string``?

**Review by McDoyen — CHANGES_REQUESTED** [review]

File: src/components/SwitchContainer.ts (hunk 2 above)
Comment (jseremba) [reply src/components/SwitchContainer.ts]: Done

File: src/components/SwitchContainer.ts (hunk 1 above)
Comment (jseremba) [reply src/components/SwitchContainer.ts]: Nanoflows can as well be used in online applications if they do not contain the following actions. 
`Create, Commit, Retrieve, Rollback`

**Pull request merged**